### DISCUSSION TITLE
Dogecoin BIP300-like sidechains or layer 2s?

### DISCUSSION THREAD

**Discussion by miller46** [dogecoin/dogecoin#3329] [General]
Curious about scaling roadmap - What is the current thinking?

Lightning? Drivechains? Fully EVM-compatible Layer2 / rollup implementations using e.g. Optimism stack?

**Comment by patricklodder** [comment]
You may want to read the discussions here: https://github.com/dogecoin/dogecoin/discussions/2264

Currently what is standing from the original proposal in that thread:
1. CSV
2. Witness segregation, but without the vBytes hack from segwit proper, because it's needed for the malleability fix of malleability source 9 listed in [bip62](https://github.com/bitcoin/bips/blob/master/bip-0062.mediawiki).

That second enhancement, especially if subsequently enriched with taproot functionality, but without is fine too, will allow for user space solutions for any L2 you mention.

I've been studying zkRollups for a while now, and I'm not nearly done, which I think can be implemented on an [Extension Block](https://github.com/litecoin-project/lips/blob/master/lip-0002.mediawiki) (much like what Litecoin did with MW). My reason for preferring the zk variant over the optimistic variant is because I believe that when we can choose between cryptography and game theory, the former gives more opportunities to harden against abuse (i.e. code > economics.)

However, I think this (rollups) is better addressed after we've (1) gotten the 1.21-dev codebase to a point where we can do a 100% compatible release, and then (2) focus on the softforks already proposed and discussed, bar any existential threats of course. If those get through, and meanwhile some more developers gained experience with how consensus changes are done, then I'm willing to put some more focus on this.
